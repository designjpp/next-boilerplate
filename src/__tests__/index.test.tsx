import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import { create } from 'react-test-renderer';

import Homepage from 'pages/index';

// import type { GetServerSidePropsContext } from 'next';

describe('<Homepage>', () => {
	it('should match snapshot', async () => {
		const component = create(
			<Homepage data={{ message: 'Message from getServerSideProps' }} />
		).toJSON();
		expect(component).toMatchSnapshot();
	});

	it('should be accessible', async () => {
		const { container } = render(
			<Homepage data={{ message: 'Message from getServerSideProps' }} />
		);
		const results = await axe(container);
		expect(results).toHaveNoViolations();
	});

	// it('should call the api and return product', async () => {
	// 	const context = { req: { headers: {} }, res: {} };
	// 	const response = await getServerSideProps(
	// 		context as GetServerSidePropsContext
	// 	);
	// 	expect(response).toEqual(
	// 		expect.objectContaining({
	// 			props: {
	// 				// @ts-ignore
	// 				...attachBrowserServerSideData(context.req),
	// 				data: {
	// 					message: 'Message from getServerSideProps',
	// 				},
	// 			},
	// 		})
	// 	);
	// });
});
