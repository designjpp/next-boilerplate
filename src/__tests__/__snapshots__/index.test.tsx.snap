// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage> should match snapshot 1`] = `
Array [
  <button
    onClick={[Function]}
  >
    Error
  </button>,
  <h1>
    Message from getServerSideProps
  </h1>,
]
`;
